java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -2634,38 +2634,120 @@ void testStringConcatWsSingleListColEmptyArrayReturnEmpty() {
-         ColumnVector result = sv.repeatStrings(1)) {
-      assertColumnsAreEqual(sv, result);
+    try (ColumnVector input = ColumnVector.fromStrings("abc", "", null, "123", null);
+         ColumnVector results = input.repeatStrings(1)) {
+      assertColumnsAreEqual(input, results);
     }
 
     // Strings column containing both null and empty.
-    try (ColumnVector sv = ColumnVector.fromStrings("abc", "", null, "123", null);
-         ColumnVector result = sv.repeatStrings( 2);
+    try (ColumnVector input = ColumnVector.fromStrings("abc", "", null, "123", null);
+         ColumnVector results = input.repeatStrings( 2);
          ColumnVector expected = ColumnVector.fromStrings("abcabc", "", null, "123123", null)) {
-      assertColumnsAreEqual(expected, result);
+      assertColumnsAreEqual(expected, results);
+    }
+  }
+
+  @Test
+  void testRepeatStringsWithColumnRepeatTimes() {
+    // Empty strings column.
+    try (ColumnVector input = ColumnVector.fromStrings("", "", "");
+         ColumnVector repeatTimes = ColumnVector.fromInts(-1, 0, 1);
+         ColumnVector results = input.repeatStrings(repeatTimes)) {
+      assertColumnsAreEqual(input, results);
+    }
+
+    // Zero and negative repeatTimes.
+    try (ColumnVector input = ColumnVector.fromStrings("abc", "xyz", "123", "456", "789", "a1");
+         ColumnVector repeatTimes = ColumnVector.fromInts(-200, -100, 0, 0, 1, 2);
+         ColumnVector results = input.repeatStrings(repeatTimes);
+         ColumnVector expected = ColumnVector.fromStrings("", "", "", "", "789", "a1a1")) {
+      assertColumnsAreEqual(expected, results);
+    }
+
+    // Strings column contains both null and empty, output is copied exactly from input.
+    try (ColumnVector input = ColumnVector.fromStrings("abc", "", null, "123", null);
+         ColumnVector repeatTimes = ColumnVector.fromInts(1, 1, 1, 1, 1);
+         ColumnVector results = input.repeatStrings(repeatTimes)) {
+      assertColumnsAreEqual(input, results);
+    }
+
+    // Strings column contains both null and empty.
+    try (ColumnVector input = ColumnVector.fromStrings("abc", "", null, "123", null);
+         ColumnVector repeatTimes = ColumnVector.fromInts(2, 3, 1, 3, 2);
+         ColumnVector results = input.repeatStrings(repeatTimes);
+         ColumnVector expected = ColumnVector.fromStrings("abcabc", "", null, "123123123", null)) {
+      assertColumnsAreEqual(expected, results);
+    }
+  }
+
+  @Test
+  void testRepeatStringsWithColumnRepeatTimesAndPrecomputedOutputSizes() {
+    // Empty strings column.
+    try (ColumnVector input = ColumnVector.fromStrings("", "", "");
+         ColumnVector repeatTimes = ColumnVector.fromInts(-1, 0, 1);
+         ColumnView.OutputSizesRepeatStrings outputSizes = input.computeOutputSizesRepeatStrings(repeatTimes)) {
+      assertEquals(0, outputSizes.totalStringSize);
+      assertNotEquals(null, outputSizes.stringSizes);
```

Comment:
I don't think it's critical to do a null check in a unit test.  The NPE itself will fail the test, since we're dereferencing the variable in the try block below.  If we do keep these null checks, might as well use the dedicated method, e.g.:
```suggestion
      assertNotNull(outputSizes.stringSizes);
```

Suggestion also applies to similar places below.